frontend/model/contracts/group.js

```diff
@@ -819,6 +819,21 @@ sbp('chelonia/defineContract', {
         archiveProposal({ state, proposalHash: data.proposalHash, proposal, contractID })
       }
     },
+    'gi.contracts/group/markProposalsExpired': {
+      validate: objectOf({
+        proposalIds: arrayOf(string)
+      }),
+      process ({ data, meta, contractID }, { state }) {
+        if (data.proposalIds.length) {
+          for (const proposalId of data.proposalIds) {
+            const proposal = state.proposals[proposalId]
+
+            Vue.set(proposal, 'status', STATUS_EXPIRED)
+            archiveProposal({ state, proposalHash: proposalId, proposal, contractID })
+          }
+        }
+      }
+    },
```

Comment:
Let's imagine 2 users were online at the same time and their clients both sent this message at the same time. In that case we'd call `archiveProposal` twice on the same proposal, even though it was already archived. That could lead to problems.

So inside here we also need to have a check to make sure that `proposal` actually exists on `state.proposals[proposalId]` (because it wouldn't exist if 2 messages were sent - the 2nd message would result in an error upon processing)